src/System.Private.CoreLib/shared/System/IO/StreamHelpers.CopyValidation.cs

```diff
@@ -42,5 +46,44 @@ public static void ValidateCopyToArgs(Stream source, Stream destination, int buf
                 throw new NotSupportedException(SR.NotSupported_UnwritableStream);
             }
         }
+
+        public static void ValidateCopyToArgs(Stream source, ReadOnlySpanAction<byte, object?> callback, int bufferSize)
+        {
+            if (callback == null)
+            {
+                throw new ArgumentNullException(nameof(callback));
+            }
+
+            ValidateCopyToArgsCore(source, bufferSize);
+        }
+
+        public static void ValidateCopyToArgs(Stream source, Func<ReadOnlyMemory<byte>, object?, CancellationToken, ValueTask> callback, int bufferSize)
+        {
+            if (callback == null)
+            {
+                throw new ArgumentNullException(nameof(callback));
+            }
+
+            ValidateCopyToArgsCore(source, bufferSize);
+        }
```

Comment:
You don't need both overloads: `callback` can just be defined as `Delegate` in a single overload, and then you can reinline the "Core" method.

Reply:
Fixed.